airflow/stats.py

```diff
@@ -250,43 +298,107 @@ def test(self, stat):
             return True  # default is all metrics allowed
 
 
+def prepare_stat_with_tags(fn: T) -> T:
+    """Add tags to stat with influxdb standard format if influxdb_tags_enabled is True."""
+
+    @wraps(fn)
+    def wrapper(self, stat=None, *args, tags=None, **kwargs):
+        if self.influxdb_tags_enabled:
+            if stat is not None and tags is not None:
+                for k, v in tags.items():
+                    if not set(",=").intersection(set(v + k)):
```

Comment:
Probably better to use the same `all((x not in y) for x in v + k)` approach; it’s much more readable than using set here.

(Or we can introduce a util function that wraps the set operation to make it easier to understand for readers)